packages/cli/src/actions/start.ts

```diff
@@ -15,10 +16,20 @@ const start: StartHandler = async (filename: string, opts: any) => {
     spinner.fail('Please specify the config path');
     return process.exit(1);
   }
-  filename = path.isAbsolute(filename) ? filename : path.join(process.cwd(), filename);
-  if (!existsSync(filename)) {
-    spinner.fail(`${filename} not exists`);
-    return process.exit(1);
+  let urlObj = url.parse(filename);
+  if (urlObj.protocol) {
+    if (urlObj.protocol !== 'http:' && urlObj.protocol !== 'https:') {
+      spinner.fail(`protocol ${urlObj.protocol} is not supported`);
+      return process.exit(1);
+    }
+  } else {
+    filename = path.isAbsolute(filename) ? filename : path.join(process.cwd(), filename);
+    if (!existsSync(filename)) {
+      spinner.fail(`${filename} not exists`);
+      return process.exit(1);
+    }
+    urlObj = url.parse(`file://${filename}`);
+    filename = urlObj.href;
```

Comment:
```js
const urlObj = url.parse(filename);
if (urlObj.protocol == null) {
  // FIXME(feely): file by default if protocol is null
  filename = path.isAbsolute(filename) ? filename : path.join(process.cwd(), filename);
  // check the filename existence
  if (!existsSync(filename)) {
    return process.exit(1);
  }
  filename = `file://${filename}`;
} else if (urlObj.protocol.indexOf([ 'http:', 'https:' ]) === -1) {
  return process.exit(1);
}
```

Reply:
`filename` should be encoded if we use file-protocol.

Reply:
Done.